tests/integration_tests/build/test_dependencies.py

```diff
@@ -4,8 +4,16 @@
 
 import os
 import ast
+
 import pytest
+
 from framework import utils
+from host_tools import proc
+
+
+pytestmark = pytest.mark.skipif(
+    "Intel" not in proc.proc_type(), reason="test only runs on Intel"
+)
```

Comment:
It may be better to use [a marker](https://docs.pytest.org/en/7.1.x/how-to/mark.html) for tests which are architecture independent like `--nonci`.
```suggestion
pytestmark = [pytest.mark.architecture_independent]
```
I would also prefer to apply it directly to tests as I think this is clearer than `pytestmark ` e.g.
```python
@pytest.mark.architecture_independent
def test_licenses():
```

Reply:
I like it, but would that stop them for running in all but one architectures though?

Reply:
I thought the meaning was that this test only runs in x86 because the list of dependencies in ARM may be different? 

If it can run in both, then we shouldn't skip it. Skipping this is not worth the complexity, IMO.

Reply:
I _think_ the meaning of this is that it doesn't matter where does it run **and** it only needs to run once.

Reply:
I agree with @pb8o that these tests don't really warrant the complexity required to skip them given how quick they are.

Reply:
well, as it is there's no complexity. It works already :P